src/libdredd/src/mutation_replace_expr.cc

```diff
@@ -0,0 +1,221 @@
+
+  int mutant_offset = 0;
+  const clang::BuiltinType* exprType =
+      expr_.getType()->getAs<clang::BuiltinType>();
+
+  if (expr_.isLValue() && ast_context.getLangOpts().CPlusPlus) {
+    new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                 << mutant_offset << ")) return ++" << arg_evaluated << ";\n";
+    mutant_offset++;
+
+    new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                 << mutant_offset << ")) return --" << arg_evaluated << ";\n";
+    mutant_offset++;
+  }
+
+  if (!expr_.isLValue() &&
+      (exprType->isBooleanType() || exprType->isInteger())) {
+    new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                 << mutant_offset << ")) return !(" << arg_evaluated << ");\n";
+    mutant_offset++;
+
+    new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                 << mutant_offset << ")) return ~(" << arg_evaluated << ");\n";
+    mutant_offset++;
+  }
+
+  // Insert constants for integer expressions
+  if (!expr_.isLValue()) {
+    if (exprType->isBooleanType()) {
+      // Replace expression with true
+      new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                   << mutant_offset << ")) return true;\n";
+      mutant_offset++;
+
+      // Replace expression with false
+      new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                   << mutant_offset << ")) return false;\n";
+      mutant_offset++;
+    }
+
+    if (exprType->isInteger() && !exprType->isBooleanType()) {
+      // Replace expression with 0
+      new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                   << mutant_offset << ")) return 0;\n";
+      mutant_offset++;
+
+      // Replace expression with 1
+      new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                   << mutant_offset << ")) return 1;\n";
+      mutant_offset++;
+    }
+
+    if (exprType->isSignedInteger()) {
+      // Replace signed integer expression with -1
+      new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                   << mutant_offset << ")) return -1;\n";
+      mutant_offset++;
+    }
+
+    if (exprType->isFloatingPoint()) {
+      // Replace floating point expression with 0.0
+      new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                   << mutant_offset << ")) return 0.0;\n";
+      mutant_offset++;
+
+      // Replace floating point expression with 1.0
+      new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                   << mutant_offset << ")) return 1.0;\n";
+      mutant_offset++;
+
+      // Replace floating point expression with -1.0
+      new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                   << mutant_offset << ")) return -1.0;\n";
+      mutant_offset++;
+    }
+  }
```

Comment:
This could all be extracted into its own method. And again this could be a static method that the other classes call.

Reply:
Done as part of the refactoring described above.